### PR TITLE
184347087 Ft user login 

### PR DESCRIPTION
**Pull request by UwaweTera**
### What does this PR do?

User login with passport

### Description of Task to be completed?

This login user can make login and receive his/her name and token to use, when provided wrong data they can accept it

### How should this be manually tested?

```
  npm install 
  npm start
```
### Any background context you want to provide?

none

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/n/projects/2628461/184347087

### Screenshots (if appropriate)

login prof

### Questions:

# Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Rutarenzi — APPROVED** [review]

**Review by NG-Solennel — APPROVED** [review]

**Review by NG-Solennel — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -17,6 +17,8 @@ DATABASE_USER= postgres #example:postgres
 DATABASE_PASSWORD=password
```
Comment (Eli250) [inline .env.example:17]: Remove values to these variables.
Eg: DATABASE_PASSWORD=
      PRODUCTION_DATABASE=
      JWT_SECRET=

File: src/controllers/userController.js [hunk 2]
```diff
@@ -38,3 +38,20 @@ export class UserController {
     }
   }
 }
+
+const userLogin = async (req, res) => {
+  try {
+    const user = {
```
Comment (Eli250) [inline src/controllers/userController.js:44]: Use object destructuring here

File: src/validation/user.vald.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Joi from "joi";
+
+const userSchema = Joi.object({
+  email: Joi.string().required().email(),
+  password: Joi.string().required(),
+});
+const userVald = async (req, res, next) => {
```
Comment (Eli250) [inline src/validation/user.vald.js:7]: Use meaningful variable name

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Review by Eli250 — CHANGES_REQUESTED** [review]
Resolve all addressed issues

File: .env.example (hunk 1 above)
Comment (Eli250) [inline .env.example:17]: Unassign these  variables too

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Comment by NG-Solennel** [comment]
you have conflicts

**Review by NG-Solennel — APPROVED** [review]

**Review by Eli250 — APPROVED** [review]

**Pull request merged**